web/components/contract-search.tsx

```diff
@@ -132,31 +147,32 @@ export function ContractSearch(props: {
     }
   }
 
+  const onSearchParametersChanged = useRef(
```

Comment:
An alternative way to do this would be to use `useState`:

```
const [onSearchParametersChanged] = useState(() => debounce(...))
```

Initializes it once in a way where it is obvious it cannot change (whereas with refs you expect mutation).

Reply:
Hmm, I don't share your intuition at all, but I will change it if you like.

I note that in other parts of our codebase we use `useMemo` or `useCallback` for this purpose, but React explicitly calls that out as being incorrect.

Reply:
Another pattern is like what you have, except you call .current immediately: `useRef().current`. That also works because there's no way to modify it.

Reply:
That makes sense to me, so if you like it I will do it.